src/overlays/actors/ovl_Bg_Spot01_Idohashira/z_bg_spot01_idohashira.c

```diff
@@ -21,49 +49,304 @@ const ActorInit Bg_Spot01_Idohashira_InitVars = {
+    return 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB504.s")
+s32 func_808AB29C(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction;
+    Vec3f* thisPos;
+    f32 endX;
+    f32 temp_f0;
+    s32 pad2;
+    Vec3f initPos;
+    f32 endZ;
+    f32 tempY;
+    f32 tempZ;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB510.s")
+    npcAction = func_808AB144(globalCtx, 2);
+    if (npcAction != NULL) {
+        temp_f0 = func_8006F93C(npcAction->endFrame, npcAction->startFrame, globalCtx->csCtx.frames);
+        initPos = this->dyna.actor.initPosRot.pos;
+        endX = npcAction->endPos.x;
+        tempY = ((kREG(10) + 1100.0f) / 10.0f) + npcAction->endPos.y;
+        endZ = npcAction->endPos.z;
+        thisPos = &this->dyna.actor.posRot.pos;
+        thisPos->x = ((endX - initPos.x) * temp_f0) + initPos.x;
+        thisPos->y =
+            func_808AB1DC(initPos.y, tempY, npcAction->endFrame, npcAction->startFrame, globalCtx->csCtx.frames) +
+            initPos.y;
+        thisPos->z = ((endZ - initPos.z) * temp_f0) + initPos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB530.s")
+        if (temp_f0 >= 1.0f) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB570.s")
+void func_808AB3E8(BgSpot01Idohashira* this) {
+    this->action = 1;
+    this->drawConfig = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/BgSpot01Idohashira_Update.s")
+void func_808AB3F8(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    this->action = 2;
+    this->drawConfig = 0;
+    this->unk_170 = 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/BgSpot01Idohashira_Init.s")
+void func_808AB414(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    func_808AAE6C(this, globalCtx);
+    this->action = 3;
+    this->drawConfig = 0;
+}
 
+void func_808AB444(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction = func_808AB144(globalCtx, 2);
+    u32 action;
+    u32 unk_16C;
+
+    if (npcAction != NULL) {
+        action = npcAction->action;
+        unk_16C = this->unk_16C;
+        if (action != unk_16C) {
+            switch (action) {
+                case 1:
+                    func_808AB3E8(this);
+                    break;
+                case 2:
+                    func_808AB3F8(this, globalCtx);
+                    break;
+                case 3:
+                    Actor_Kill(&this->dyna.actor);
+                    break;
+                default:
+                    osSyncPrintf("Bg_Spot01_Idohashira_Check_DemoMode:そんな動作は無い!!!!!!!!\n");
+            }
+            this->unk_16C = action;
```

Comment:
16C can be named npcAction. I would also name the temp in this function "npcAction" as to not confuse it with the actionFunc index "action"

Reply:
There's already an npcAction temp in this function, which is the CsCmdActorAction* - I'll call the unk_16C thisNpcAction.